freqtrade/freqai/freqai_interface.py

```diff
@@ -354,25 +357,43 @@ def start_backtesting(
                         logger.warning(
                             f"Training {pair} raised exception {msg.__class__.__name__}. "
                             f"Message: {msg}, skipping.", exc_info=True)
+
+                        if self.new_feature_selector:
```

Comment:
What is the reason for putting this behind a flag? As far as I can tell, the proposed skip feature would be an improvement, so I don't see why anyone would opt out of it. Unless I am missing something?

Reply:
I did this as a flag because training on data with gaps is not standard behaviour, usually for training there should be all data. A more stable model is built on a stable dataset with no skips - then the usual failure (as in a train) will occur. For the more complex case, options are provided for the advanced user.

Reply:
It appears that there is some confusion of how `start_backtesting()` works. Let me try to explain it in plain english:

It:
0) gets a full `--timerange` of data for a single coin from the strategy.
1) splits the `--timerange` up according to `--timerange` / `backtest_period_days` (the user desired backtest timerange divided by the `backtest_period_days` parameter.) Let's say this ends up being 10 sub-timeranges.
2) for each sub timerange, it trains a model.
3) It uses that model make an inference on the subsequent `backtest_period_days`. 
4) It appends these predictions to the return dataframe. 
5) move to next timerange and repeat steps 2-4. 
6) do this until there are no more timeranges.

This means that it is training X separate models, where X is `--timerange`/`backtest_period_days`

The issue that your PR is aiming to alleviate is to allow one of those model trainings to fail. That may be due to lack of data or may be due to some other problem in the training procedure. 

So that means that there will be no model trained with "gaps" in data as you are suggesting. The only thing that occurs is if one of the sub timerange model trainings fails, then that entire sub timerange `backtest_period_days` will return some filler + `do_predict = 0` so that the user strategy doesnt act on anything for that sub timerange.

This has nothing to do with a model getting trained on data gaps. Therefore, there is no reason to be hiding this functionality behind a flag.